index.js

```diff
@@ -1,149 +1,38 @@
-  },
-  vega: {
-    custom: (path, query, host, proto) => {
-      if (!path.match(/\/search\/X/i)) { return null; }
-      let searchKey = Object.keys(query).find(key => key.match(/search/i));
-      if (!searchKey) { return null; }
-      let searchValue = query[searchKey];
-      if (!Array.isArray(searchValue) || !searchValue[0] || (typeof searchValue[0] !== 'string')) { return null; }
-      return searchValue[0].match(/t:\((.*)\)(?:%20|\s*)and(?:%20|\s*)a:\((.*)\)/i)
-    },
-    handler: match => `${BASE_SCC_URL}/search?contributor=${match[2]}&title=${match[1]}`
-  },
-  oclc: {
-    expr: /\/search\/o\=?(\d+)/,
-    handler: match => `${BASE_SCC_URL}/search?oclc=${match[1]}&redirectOnMatch=true`,
-  },
-  issn: {
-    expr: /\/search\/i(\d{4}\-\d{4})/,
-    handler: match => `${BASE_SCC_URL}/search?issn=${match[1]}&redirectOnMatch=true`,
-  },
-  isbn: {
-    expr: /\/search\/i(\w+)/,
-    handler: match => `${BASE_SCC_URL}/search?isbn=${match[1]}&redirectOnMatch=true`,
-  },
-  searchRegWith: {
-    expr: /\/search(~S\w*)?\/([a-zA-Z])(([^\/])+)/,
-    handler: match => `${BASE_SCC_URL}/search?q=${recodeSearchQuery(match[3])}${getIndexMapping(match[2])}`
-  },
-  searchRegWithout: {
-    expr: /\/search(~S\w*)?(\/([a-zA-Z]))?/,
-    handler: (match, query) => {
-      const mappedQuery = getQueryFromParams(match[0], query);
-      if (!mappedQuery) return BASE_SCC_URL;
-      return `${BASE_SCC_URL}/search?q=${mappedQuery}${getIndexMapping(match[3])}`;
-    }
-  },
-  patroninfoReg: {
-    expr: /^\/patroninfo/,
-    handler: match => `${BASE_SCC_URL}/account`
-  },
-  recordReg: {
-    expr: /\/record=(\w+)/,
-    handler: (match) => {
-      const bnum = match[1];
-      return `${BASE_SCC_URL}/bib/${bnum}`;
-    }
-  },
-  legacyReg: {
-    expr: /pinreset|selfreg/,
-    handler: (match, query) => {
-        return `${LEGACY_CATALOG_URL}${match.input}${reconstructQuery(query)}`
-    }
-  },
-};
-
-function reconstructQuery(query) {
-  const reconstructedQuery = Object.entries(query).map(([key, values]) => {
-      return values.map(value => value.length ? `${key}=${value}` : key).join('&')
-    })
-    .join('&');
-  return reconstructedQuery.length ? `?${reconstructedQuery}` : ''
-}
-
-function reconstructOriginalURL(path, query, host, proto) {
-  const reconstructedQuery = Object.entries(query).map(([key, values]) => {
-      return values.map(value => value.length ? `${key}=${value}` : key).join('&')
-    })
-    .join('&');
-  return encodeURIComponent(`${proto}://${host}${path}${reconstructQuery(query)}`);
-}
-
 // The main method to build the redirectURL based on the incoming request
 // Given a path and a query, finds the first expression declared above which matches
 // the path, and returns the corresponding handler with the matchdata and query
 // As a default, returns the BASE_SCC_URL
-function mapWebPacUrlToSCCURL(path, query, host, proto) {
+function mapToRedirectURL (path, query, host, proto) {
```

Comment:
Thank you for changing the method name to keep it meaningful! I appreciate that effort.